tests/test_blockdiag.py

```diff
@@ -0,0 +1,49 @@
+from mpi4py import MPI
+import numpy as np
+from numpy.testing import assert_allclose
+np.random.seed(42)
+
+import pytest
+
+import pylops
+import pylops_mpi
+
+par1 = {'ny': 101, 'nx': 101, 'dtype': np.float64}
+par1j = {'ny': 101, 'nx': 101, 'dtype': np.complex128}
+par2 = {'ny': 301, 'nx': 101, 'dtype': np.float64}
+par2j = {'ny': 301, 'nx': 101, 'dtype': np.complex128}
+
+
+@pytest.mark.mpi(min_size=2)
+@pytest.mark.parametrize("par", [(par1), (par1j), (par2), (par2j)])
+def test_blockdiag(par):
+    size = MPI.COMM_WORLD.Get_size()
+    rank = MPI.COMM_WORLD.Get_rank()
+    Op = pylops.MatrixMult(A=((rank + 1) * np.ones(shape=(par['ny'], par['nx']))).astype(par['dtype']))
+    BDiag_MPI = pylops_mpi.MPIBlockDiag(ops=[Op, ])
+
+    x = pylops_mpi.DistributedArray(global_shape=size * par['nx'], dtype=par['dtype'])
+    x[:] = np.ones(shape=par['nx'], dtype=par['dtype'])
+    x_global = x.asarray()
+
+    y = pylops_mpi.DistributedArray(global_shape=size * par['ny'], dtype=par['dtype'])
+    y[:] = np.ones(shape=par['ny'], dtype=par['dtype'])
+    y_global = y.asarray()
+
+    x_mat = BDiag_MPI._matvec(x)
```

Comment:
I did use `_matvec` in my quick tests because I realized we could not use ``matvec / @` without making a new LinearOperator, but ideally we should include it in this PR and then never use internal methods like `_matvec ` in tests :)

Reply:
Changed it from `_matvec` -> `@`
Changed it from `_rmatvec` -> `.H @`